### PR TITLE
feat(react-components): add support for system data models to react components

### PR DESCRIPTION
**Pull request by christjt**
#### Type of change

Feat  

## Description :pencil:

Uses system data models instead of FDM config.

Many thing are still suboptimal but adding a PR to be able to move forward.

- Styling is not cached, i.e. when anything changes we flush everything.
- Highlighting seems bugged now (style flushing is very visible, and there seems to be an issue in reveal causing clicked objects to suddanly dissappear, but it comes back if you move the camera a little)
- The fetching of the mapped nodes should be an infinite query such that partial styling can be applied as you go.
- Missing the ability to scope mapped nodes.

## Test instructions :information_source:

Model styling story and highlighting story have been updated.

## Checklist :ballot_box_with_check:

- [ ] I am proud of this feature.
- [ ] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/components/Reveal3DResources/queryMappedData.ts [hunk 1]
```diff
@@ -91,26 +86,25 @@ async function getAncestorNodesForTreeIndex(
...
   const filter = {
     and: [
       {
         equals: {
           property: ['edge', 'endNode'],
           value: {
-            space: fdmConfig.global3dSpace,
-            externalId: `model_3d_${model.modelId}`
+            space: INSTANCE_SPACE_3D_DATA,
+            externalId: `${model.modelId}`
```
Comment (haakonflatval-cognite) [inline react-components/src/components/Reveal3DResources/queryMappedData.ts:99]: Have we adopted this new convention in naming 3D models in FDM? Sounds good to me

File: react-components/src/components/Reveal3DResources/Reveal3DResources.tsx [hunk 2]
```diff
@@ -67,16 +51,12 @@ export const Reveal3DResources = ({
     getTypedModels(resources, viewer).then(setReveal3DModels).catch(console.error);
   }, [resources, viewer]);
 
-  const modelsStyling = useCalculateModelsStyling(reveal3DModels, styling, fdmAssetMappingConfig);
-
-  useEffect(() => {
-    setReveal3DModelsStyling(modelsStyling);
-  }, [modelsStyling]);
+  const reveal3DModelsStyling = useCalculateModelsStyling(reveal3DModels, instanceStyling ?? []);
```
Comment (haakonflatval-cognite) [inline react-components/src/components/Reveal3DResources/Reveal3DResources.tsx:54]: This variable doesn't seem to be used anywhere, is it necessary?

**Review by haakonflatval-cognite — CHANGES_REQUESTED** [review]
Some minor change requests - but otherwise looks fine :+1:

File: react-components/src/components/Reveal3DResources/Reveal3DResources.tsx (hunk 2 above)
Comment (christjt) [reply react-components/src/components/Reveal3DResources/Reveal3DResources.tsx:54]: It is used further down and passed to appropriate models.

File: react-components/src/components/Reveal3DResources/queryMappedData.ts (hunk 1 above)
Comment (christjt) [reply react-components/src/components/Reveal3DResources/queryMappedData.ts:99]: Backend has changed the convention, yes.

File: react-components/stories/HighlightNode.stories.tsx [hunk 3]
```diff
@@ -32,37 +33,43 @@ export const Main: Story = {
...
+        <CameraController
+          initialFitCamera={{
+            to: 'allModels'
+          }}
+          cameraControlsOptions={{
+            changeCameraTargetOnClick: true,
+            mouseWheelAction: 'zoomToCursor'
+          }}
+        />
+        <ReactQueryDevtools />
```
[the remark this answers is not in the document]
Comment (christjt) [reply react-components/stories/HighlightNode.stories.tsx:62]: Think it makes sense here to be able to debug react queries that happen on click.

**Pull request merged**